modules/arkit/arkit_interface.mm

```diff
@@ -162,37 +172,42 @@
 }
 
 Array ARKitInterface::raycast(Vector2 p_screen_coord) {
-	Array arr;
-	Size2 screen_size = OS::get_singleton()->get_window_size();
-	CGPoint point;
-	point.x = p_screen_coord.x / screen_size.x;
-	point.y = p_screen_coord.y / screen_size.y;
-
-	///@TODO maybe give more options here, for now we're taking just ARAchors into account that were found during plane detection keeping their size into account
-	NSArray<ARHitTestResult *> *results = [ar_session.currentFrame hittest:point types:ARHitTestResultTypeExistingPlaneUsingExtent];
-
-	for (ARHitTestResult *result in results) {
-		Transform transform;
-
-		matrix_float4x4 m44 = result.worldTransform;
-		transform.basis.elements[0].x = m44.columns[0][0];
-		transform.basis.elements[1].x = m44.columns[0][1];
-		transform.basis.elements[2].x = m44.columns[0][2];
-		transform.basis.elements[0].y = m44.columns[1][0];
-		transform.basis.elements[1].y = m44.columns[1][1];
-		transform.basis.elements[2].y = m44.columns[1][2];
-		transform.basis.elements[0].z = m44.columns[2][0];
-		transform.basis.elements[1].z = m44.columns[2][1];
-		transform.basis.elements[2].z = m44.columns[2][2];
-		transform.origin.x = m44.columns[3][0];
-		transform.origin.y = m44.columns[3][1];
-		transform.origin.z = m44.columns[3][2];
-
-		/* important, NOT scaled to world_scale !! */
-		arr.push_back(transform);
-	}
+	if (@available(iOS 11, *)) {
```

Comment:
```suggestion
	if (!@available(iOS 11, *)) {
		return Array();
	}

	Array arr;
	...
```
That saves an extra indentation and a lot of diff. Same for other methods which follow the same pattern (early return instead of big `if`).

Reply:
Using negative check for `available` might not work (I think I have tried that), but I'll look into it.

Reply:
Checks like `if (@available(..) || something)` do not compile, it's probably same with the `not`.